datadog/fwprovider/resource_datadog_csm_resource_evaluation_filter.go

```diff
@@ -0,0 +1,293 @@
+package fwprovider
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+type CsmResourceEvaluationFilter struct {
+	API  *datadogV2.SecurityMonitoringApi
+	Auth context.Context
+}
+
+type ResourceEvaluationFilterModel struct {
+	CloudProvider types.String `tfsdk:"cloud_provider"`
+	ID            types.String `tfsdk:"id"`
+	Tags          types.List   `tfsdk:"tags"`
+}
+
+func NewResourceEvaluationFilter() resource.Resource {
+	return &CsmResourceEvaluationFilter{}
+}
+
+var (
+	_ resource.ResourceWithConfigure   = &CsmResourceEvaluationFilter{}
+	_ resource.ResourceWithImportState = &CsmResourceEvaluationFilter{}
+)
+
+func (r *CsmResourceEvaluationFilter) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	providerData, _ := request.ProviderData.(*FrameworkProvider)
+	r.API = providerData.DatadogApiInstances.GetSecurityMonitoringApiV2()
+	r.Auth = providerData.Auth
+}
+
+func (r *CsmResourceEvaluationFilter) Metadata(_ context.Context, _ resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "resource_evaluation_filter"
+}
+
+var tagFormatValidator = stringvalidator.RegexMatches(
+	regexp.MustCompile(`^[^:]+:[^:]+$`),
+	"each tag must be in the format 'key:value' (colon-separated)",
+)
+
+func toSliceString(list types.List) ([]string, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	result := make([]string, 0)
+
+	if list.IsNull() || list.IsUnknown() {
+		return result, nil
+	}
+
+	for _, elem := range list.Elements() {
+		strVal, ok := elem.(types.String)
+		if !ok {
+			diags.AddError("Invalid element type", "Expected string in list but found a different type")
+			continue
+		}
+		result = append(result, strVal.ValueString())
+	}
+
+	return result, diags
+}
+
+func (r *CsmResourceEvaluationFilter) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manage a single resource evaluation filter.",
+		Attributes: map[string]schema.Attribute{
+			"cloud_provider": schema.StringAttribute{
+				Required:    true,
+				Description: "The cloud provider of the filter's targeted resource.",
```

Comment:
We only support three values here. Can we provide this in the description?

I'm guessing it will be 'aws', 'gcp' or 'azure'